src/core/thread/link_metrics.hpp

```diff
@@ -0,0 +1,162 @@
+/*
+ *  Copyright (c) 2020, The OpenThread Authors.
+ *  All rights reserved.
+ *
+ *  Redistribution and use in source and binary forms, with or without
+ *  modification, are permitted provided that the following conditions are met:
+ *  1. Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *  2. Redistributions in binary form must reproduce the above copyright
+ *     notice, this list of conditions and the following disclaimer in the
+ *     documentation and/or other materials provided with the distribution.
+ *  3. Neither the name of the copyright holder nor the
+ *     names of its contributors may be used to endorse or promote products
+ *     derived from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ *  AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ *  IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ *  ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ *  LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ *  CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ *  SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ *  INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ *  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ *  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/**
+ * @file
+ *   This file includes definitions for Thread Link Metrics query and management.
+ */
+
+#ifndef LINK_METRICS_HPP_
+#define LINK_METRICS_HPP_
+
+#include "openthread-core-config.h"
+
+#if OPENTHREAD_CONFIG_MLE_LINK_METRICS_ENABLE
+
+#include <openthread/ip6.h>
+
+#include "common/code_utils.hpp"
+#include "common/locator.hpp"
+
+#include "link_metrics_tlvs.hpp"
+#include "topology.hpp"
+
+namespace ot {
+
+/**
+ * @addtogroup core-link-metrics
+ *
+ * @brief
+ *   This module includes definitions for Thread Link Metrics query and management.
+ *
+ * @{
+ */
+
+class LinkMetrics : public InstanceLocator
+{
+public:
+    /**
+     * This constructor initializes an instance of the LinkMetrics class.
+     *
+     * @param[in]  aInstance  A reference to the OpenThread interface.
+     *
+     */
+    explicit LinkMetrics(Instance &aInstance);
+
+    /**
+     * This function sends an MLE Data Request containing Link Metrics Query TLV to query Link Metrics data. It could
```

Comment:
Brief should be a single sentence with empt line after followed by detailed description.